pandas/core/groupby/ops.py

```diff
@@ -490,21 +495,26 @@ def _call_cython_op(
         orig_values = values
 
         dtype = values.dtype
-        is_numeric = is_numeric_dtype(dtype)
+        dtype_kind = dtype.kind
+        # is_numeric_dtype
+        is_numeric = dtype_kind in "uifcb"
 
-        is_datetimelike = needs_i8_conversion(dtype)
+        # is_datetimelike = needs_i8_conversion(dtype)
+        is_datetimelike = dtype_kind in ["m", "M"] or (
+            dtype_kind == "O" and dtype.type is Period
```

Comment:
at this point we have an ndarray, so PeriodDtype shouldn't be possible i think?

Reply:
@jorisvandenbossche can you respond to a few small comments here?  this should be pretty easy to merge and will be a nice perf bump